src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs

```diff
@@ -1212,9 +1214,12 @@ private void InvalidateTextBlock()
 
 		public override string GetAccessibilityInnerText() => Text;
 
-		// This approximates UWP behavior
-		private protected override double GetActualWidth() => DesiredSize.Width;
-		private protected override double GetActualHeight() => DesiredSize.Height;
+		// On UWP, ActualSize is immediately available after the TextBlock is measured, even if the TextBlock
+		// is never added to the visual tree. In Uno, we just return a best effort size until the TextBlock is
+		// attached to the visual tree and then we can depend on the layout cycle to return the proper ActualSize
+		private protected override double GetActualWidth() => _textDesiredSize is not null ? _textDesiredSize.Value.Width : Padding.Left + Padding.Right;
```

Comment:
Why is `Padding` not being added when `_textDesiredSize` is not null?

Reply:
I'm not sure whether that's the behavior on Windows, needs to be checked

Reply:
_textDesiredSize already includes the padding actually.